contrib/test_jenkins.sh

```diff
@@ -900,18 +900,57 @@ run_ucx_perftest() {
 			do
 				for gdr in $gdr_options
 				do
-					if [ $with_mpi -eq 1 ]
+					if [ "X$tls" == "Xcuda_ipc" ]
```

Comment:
You also need to add `cuda_ipc` to the `tls_list` in https://github.com/openucx/ucx/blob/8bb8372568b8969ed066030fd9dbf2896c1a5140/contrib/test_jenkins.sh#L878

Reply:
@pentschev Added the specific test you gave https://github.com/openucx/ucx/pull/4476/files#diff-b0527020d65dad8a6357eade47e32726R932

but with a list of large message sizes
https://github.com/openucx/ucx/pull/4476/files#diff-fc681478b33c9debf437b8e424382027R1

with cache explicitly disabled. Can you provide a review in case I've missed something?

Reply:
That said, I think another change is still needed to ensure that we don't pick non-peer-accessible GPUs, otherwise the tests will take a really long time to finish. Alternatively, if there's just one GPU or no peer-accessible GPUs we need to disable this test.

Reply:
It seems currently Mellanox testing node has only one GPU/node.  So both ranks will pick the same GPU and cuda-IPC accessible.

Reply:
> It seems currently Mellanox testing node has only one GPU/node. So both ranks will pick the same GPU and cuda-IPC accessible.

This is true if GPUs are set to MPS mode and not exclusive mode. @bureddy Can you confirm if that's the case? 

Ideally we shouldn't have to do this because for HPC at least, exclusive mode is the common case. Is it possible to reconfigure the nodes to have more than a single GPU to test cuda-ipc in exclusive mode?

Reply:
they are in default mode, not in exclusive mode

Reply:
So, cuda-ipc should test that we added should be tested. 

@bureddy can you also post the size of physical memory on the GPU? Right now, I test with as much as 2GB for each process, so a total of at least 4GB is needed. I can reduce this limit if the GPU capacity is lesser than that.

Reply:
it is 16 MB.  But the problem is multiple gpu test workers and multiple PRs might be in testing in parallel

Reply:
16 GB?

Reply:
yes

Reply:
16 MB seems too low.

Reply:
its is 16 GB

Reply:
Thanks for adding the tests @Akshay-Venkatesh, it looks good to me overall, but since we're adding tests for disabled cache, wouldn't it be useful to also test with cache enabled as well? I think there are no other tests for CUDA IPC cache, so the same tests you added could be useful for that.

If the tests take really long, I think we don't really need to test all those sizes though, maybe something like 4MB, 128MB, 1024MB and 2048MB would suffice.

Reply:
@pentschev Added ipc_cache toggle to the test but keeping different sizes.

@bureddy I checked if the test works with a single GPU in default mode and it does. As long as there is at least 1 GPU, the test will be triggered.

Reply:
Thanks @Akshay-Venkatesh !